contracts/scripts/deploy.ts

```diff
@@ -84,22 +124,24 @@ const deployAll = async () => {
         pedersenHash.address
       );
 
+      // TEST STUB FOR LOCAL TESTING
       const StarknetCoreContractStub = await ethers.getContractFactory(
         "StarknetCoreContractStub"
       );
       starknetCoreContractStub = await StarknetCoreContractStub.deploy();
       await starknetCoreContractStub.deployed();
-
       console.log(
         "StarknetCoreContractStub contract has been deployed to: ",
         starknetCoreContractStub.address
       );
 
+      //L1Resolver deployment
       const proofProxy = await upgrades.deployProxy(
         SNResolverStub,
-        [pedersenHash.address, starknetCoreContractStub.address, ["https://localhost:8080/{sender}/{data}.json"], '0x7412b9155cdb517c5d24e1c80f4af96f31f221151aab9a9a1b67f380a349ea3'],
+        [pedersenHash.address, poseidon3.address
+          , starknetCoreContractStub.address, ["https://localhost:8080/{sender}/{data}.json"], '0x7412b9155cdb517c5d24e1c80f4af96f31f221151aab9a9a1b67f380a349ea3'],
```

Comment:
we can use the constant `L2_RESOLVER_ADDRESS` here

Reply:
This applies to all L2_RESOLVER_ADDRESS mentions below as well. They cannot all point to the same const var because frontend, contracts, gateway, client test are all independent apps with their package.json and also they happen to be the same values right now but they can be different especially in tests. Also you might want to deploy different resolvers but not yet ready to have the change reflected in the frontend app. Though I have added constants in deploy.ts file just to make it more explicit.